File: Services/Negwork.Services.Data/CommentsService.cs
namespace Negwork.Services.Data
{
    using System;
    using Contracts;
    using Negwork.Data.Models;
    using Negwork.Data.Repositories;

    public class CommentsService : ICommentsService
    {
        private IRepository<Comment> comments;

        public CommentsService(IRepository<Comment> comments)
        {
            this.comments = comments;
        }

        public Comment CreateComment(string userId, string content, int articleId, DateTime? creationDate)
        {
            var comment = new Comment()
            {
                ArticleId = articleId,
                UserId = userId,
                Content = content,
                CreationDate = creationDate
            };

            this.comments.Add(comment);
            this.comments.SaveChanges();

            return comment;
        }
    }
}

File: Client/Negwork.Client.App/scripts/app/comments/comments-service.js
(function () {
    'use strict';

    var commentsService = function commentsService(data) {
        var COMMENTS_API_URL = '/api/comments';

        function add(comment) {
            return data.post(COMMENTS_API_URL, comment);
        }

        return {
            add: add
        };
    }

    angular
        .module('negwork.services')
        .factory('comments', ['data', commentsService])
}());